src/Microsoft.TestPlatform.Extensions.BlameDataCollector/NetClientHangDumper.cs

```diff
@@ -0,0 +1,89 @@
+                    ConsoleOutput.Instance.Information(false, "Blame: Dumping this process tree (from bottom):");
+
+                    foreach (var p in processTree.OrderBy(t => t.Level))
+                    {
+                        EqtTrace.Verbose($"NetClientHangDumper.Dump: {(p.Level != 0 ? " + " : " > ")}{new string('-', p.Level)} {p.Process.Id} - {p.Process.ProcessName}");
+                        ConsoleOutput.Instance.Information(false, $"Blame: {(p.Level != 0 ? " + " : " > ")}{new string('-', p.Level)} {p.Process.Id} - {p.Process.ProcessName}");
+                    }
+                }
+                else
+                {
+                    EqtTrace.Verbose($"NetClientHangDumper.Dump: Dumping {process.Id} - {process.ProcessName}.");
+                    ConsoleOutput.Instance.Information(false, $"Blame: Dumping {process.Id} - {process.ProcessName}");
+                }
+            }
+
+            var bottomUpTree = processTree.OrderByDescending(t => t.Level).Select(t => t.Process);
+
+            foreach (var p in bottomUpTree)
+            {
+                try
+                {
+                    p.Suspend();
+                }
+                catch (Exception ex)
+                {
+                    EqtTrace.Error($"NetClientHangDumper.Dump: Error suspending process {p.Id} - {p.ProcessName}: {ex}.");
+                }
+            }
+
+            foreach (var p in bottomUpTree)
+            {
+                try
+                {
+                    var outputFile = Path.Combine(outputDirectory, $"{p.ProcessName}_{p.Id}_{DateTime.Now:yyyyMMddTHHmmss}_hangdump.dmp");
+                    EqtTrace.Verbose($"NetClientHangDumper.CollectDump: Selected dump type {type}. Dumping {process.Id} - {process.ProcessName} in {outputFile}. ");
+
+                    var client = new DiagnosticsClient(p.Id);
+
+                    // Connecting the dump generation logging to verbose output to avoid changing the interfaces again -> EqtTrace.IsVerboseEnabled
+                    // before we test this on some big repo.
+                    client.WriteDump(type == DumpTypeOption.Full ? DumpType.Full : DumpType.Normal, outputFile, logDumpGeneration: false);
+                }
+                catch (Exception ex)
+                {
+                    EqtTrace.Error($"NetClientHangDumper.Dump: Error dumping process {p.Id} - {p.ProcessName}: {ex}.");
+                }
+            }
+
+            foreach (var p in bottomUpTree)
+            {
+                try
+                {
+                    EqtTrace.Verbose($"NetClientHangDumper.Dump: Killing process {p.Id} - {p.ProcessName}.");
+                    p.Kill();
+                }
+                catch (Exception ex)
+                {
+                    EqtTrace.Error($"NetClientHangDumper.Dump: Error killing process {p.Id} - {p.ProcessName}: {ex}.");
+                }
+            }
```

Comment:
Would not be better to place it in separate method?

Reply:
Why?

Reply:
For better refactoring :)

Reply:
Refactor for better refactoring?

Reply:
To have better refactored code.

Reply:
Yes, but how adding a method helps the code to be better? It invokes a single method in between logging. This imho reads okay, and the log will be easier to parse.